### PR TITLE
Changes for generic 6502 support

### PR DESCRIPTION
**Pull request by robmcmullen**
OK, here's my first working generic 6502 support, but this is the easiest way to show you what I've done and you don't have to consider this a real pull request. The Makefile is a bit hacked; I have a python utility to manipulate DOS 3.3 filesystems (not ProDOS; too complicated for my purpsose, still need AppleCommander for that) and create DOS 3.3 binary files from data on the host filesystem, so that's that ``atrcopy`` line. I need to clean that up and announce it, which I will do soon. 

* added conditional compilation for 65C02/6502 in graphics.s, hisprite.s, macros.s
* added code generation for conditional compilation in HiSprite.py
* added argparse to HiSprite.py
* added Makefile to generate alternate build for 6502/DOS 3.3 image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,24 +9,30 @@
...
+hisprite:
+	$(CL65) -t apple2enh --start-addr $(ADDR) -l$(PGM).lst -o $(PGM) $(PGM).s
 	java -jar $(AC) -d $(PGM).dsk $(PGM)
 	java -jar $(AC) -p $(PGM).dsk $(PGM) BIN 0x$(ADDR) < $(PGM)
-	rm -f $(PGM)
-	rm -f $(PGM).o
-	osascript V2Make.scpt $(PROJECT_DIR) $(PGM)
+	#rm -f $(PGM)
+	#rm -f $(PGM).o
+	#osascript V2Make.scpt $(PROJECT_DIR) $(PGM)
```
Comment (blondie7575) [inline Makefile]: Why is the osascript commented out? This is the part that launches the emulator and inserts/runs the disk image. It's kinda the magic of this one-click build pipeline. :)

File: hisprite.s [hunk 2]
```diff
@@ -66,8 +68,8 @@ SPRITEPTR_H		= $1c
 main:
 	jsr EnableHires
 
-	lda #$00
-	jsr VenetianFill
+	;lda #$00
+	;jsr VenetianFill
```
Comment (blondie7575) [inline hisprite.s]: Why is the screen-clear commented out?

File: hisprite.s [hunk 3]
```diff
@@ -183,7 +190,7 @@ storeAndContinue:
 
 resetX:
 	lda #0
-	bra storeAndContinue
+	beq storeAndContinue
```
Comment (blondie7575) [inline hisprite.s]: Note that this sample app has changed a lot recently, so these changes may be moot and/or won't merge well.

**Comment by blondie7575** [comment]
I'm a fan of your changes- thanks for the II+ support! It won't merge at the moment, though.

File: Makefile (hunk 1 above)
Comment (robmcmullen) [reply Makefile]: sorry, checked that in by mistake.

File: hisprite.s (hunk 2 above)
Comment (robmcmullen) [reply hisprite.s]: I can't build the ProDOS version so I loaded some data at $2000 outside of the demo program so I could see the sprites moving over an image

**Comment by robmcmullen** [comment]
I haven't updated from your master in a while; I'll pull it down and reapply the changes (minus the problems above!)

**Comment by robmcmullen** [comment]
Yay! Finally mergeable. Are you are still in progress with some changes? Using the master branch (before applying my changes) the sprites don't seem to move any more.

**Comment by blondie7575** [comment]
Yah, as I mentioned earlier, this is still verrrrrry much a WIP. I would expect all this code to change rapidly over the next few weeks. I really wasn’t expecting anyone to try and open PRs on it so soon. :)

I would avoid the sample app completely- it's my testing ground for many parts of the system, and should not be expected to function or be stable at any given moment.

**Comment by robmcmullen** [comment]
Sure, no problem. I didn't really think about the distinction before, but I'll stay out of all the .s files. How about you ignore this pull request and I'll create another one with just the changes to HiSprite.py?

**Comment by blondie7575** [comment]
Closing in favor of PR #4 with just the Python changes. Thanks!

**Pull request closed without merging**